core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/configuration/PropertiesProvider.java

```diff
@@ -40,25 +41,31 @@
  * 
  * @author Leon Kiefer
  */
-public class PropertiesProvider implements ServiceFunction<Properties> {
+public class PropertiesProvider implements ServiceProvider<Properties> {
+
+	private ServiceConsumer<ConfigurationLoader> consumer;
+
+	public PropertiesProvider() {
+		this.consumer = ConsumerFactory.build(null, Util.serviceDescriptionFor(ConfigurationLoader.class));
+	}
 
 	@Override
-	public Properties getService(Map<ServiceDescription<?>, ServiceFactory<?>> resolvedDependencies,
+	public Properties getService(Map<ServiceConsumer<?>, ServiceFactory<?>> resolvedDependencies,
```

Comment:
java doc comment missing

Reply:
the javadoc is in the interface

Reply:
link it, and comment on the implementation. We need more comments in my opinion.

Reply:
`@Override`